src/filter.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    Filter([f])
+
+Uses a function `f` to identify values, rows, columns or slices of data that should be
+removed during an `apply` call. The default function `f` will removing `missing`s, or any
+rows, columns or slices containing `missing`s.
+"""
+struct Filter
+    func::Function
```

Comment:
```suggestion
struct Filter{F}
    func::F
```

We really want to let the compiler specialize on this, don't we?
Otherwise every call to `.func` will be type unstable